src/Main.cs

```diff
@@ -117,5 +132,24 @@ private void Update() {
 				uiOpen = !uiOpen;
 			}
 		}
+
+		private static void SetReadyState(MultiplayerRoomPlayer player, bool readyState)
+		{
+			player.SetReadyToBegin(readyState); 
+			MultiplayerRoomManager roomManager = (MultiplayerRoomManager)NetworkManager.singleton;
+			roomManager.ReadyStatusChanged();
+		}
+		
+		private void BanPlayer(MultiplayerRoomPlayer player)
+		{
+			bannedPlayers.Add((player.netId, player.playerName));
+			KickPlayer(player);
+		}
+
+		private static void KickPlayer(MultiplayerRoomPlayer player) =>
+			player.GetComponent<NetworkIdentity>().connectionToClient.Disconnect();
+		
+		private bool IsPlayerBanned(MultiplayerRoomPlayer player) =>
+			bannedPlayers.Any(ban => ban.Item1 == player.netId) || bannedPlayers.Any(ban => ban.Item2 == player.playerName);
```

Comment:
I would bet money that netId doesn't stay the same, and name also isn't reliable

Mirror's `Transport::ServerGetClientAddress` converts a connection ID to a SteamID with the FizzyFacepunch transport the game uses, so use that for a good identifier. Use `MultiplayerRoomManager::GetTransport` to get that transport object, earlier in the Main file I get it using the singleton

Also run clang-format

Reply:
I looked in the source code, netid is a really bad idea, but the nickname rarely changes